.github/workflows/build+test.yml

```
@@ -20,6 +20,14 @@ jobs:
       with:
         components: rustfmt
     - run: cargo fmt --all -- --check
+  
+  check:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@stable
```

```suggestion
    - uses: dtolnay/rust-toolchain@stable
    - uses: Swatinem/rust-cache@v2
```
Actually, can you add the cache here?